src/main/java/nextstep/ladder/domain/Line.java

```diff
@@ -0,0 +1,55 @@
+package nextstep.ladder.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Line {
+    private List<Point> points;
+
+    public Line(List<Point> points) {
+        this.points = points;
+        checkOverLabLine();
+    }
+
+    public Line(int countOfPerson) {
+        this.points = new ArrayList<>(countOfPerson);
+
+        points.add(new Point(RandomLine::getRandomValue));
+
+        for (int i = 1; i < countOfPerson - 1; i++) {
+            int prevIndex = i - 1;
+            points.add(new Point(() -> isBeforeDrawLine(prevIndex)));
```

Comment:
```suggestion
            boolean finalCurrentPoint = isCurrentPoint;
            Point currentPoint = new Point(() -> isBeforeDrawLine(finalCurrentPoint));
            ...
```

`i-1`같은 표현은 복잡한것 같아요.
이전의 상태를 기억하는 형태는 어때요.
결국 원하는것은 이전의 bool아닌가요?